### ISSUE TITLE
OpenSSL Verify Depth is hardcoded at 1

### ISSUE THREAD

**Issue by paultag** [unbit/uwsgi#1563]
`core/ssl.c` hardcodes `SSL_CTX_set_verify_depth` to `1`, which wont work when client Certificates are signed by an online intermediary, chained to an offline root Certificate.

I suspect this comes from some code yanked from `nginx` maybe (they seem to have the same default), which -- fair enough, it's just a bit low for a lot of medium sized Certificate deployments.

**Comment by paultag** [comment]
To start something to talk about, and in an effort to be productive, I've filed https://github.com/unbit/uwsgi/pull/1564 bumping this to 3, which should be large enough to get someone who has this as an actual problem to pony up some developer time :)

**Comment by paultag** [comment]
Curiously, however, when I used my Intermediary Certifciates in the CA Certificate file, it still rejected my Client. I wonder if OpenSSL has to have the complete chain, even if a Certificate is in the trust store, but signed by another CA

**Comment by paultag** [comment]
For future people googling this problem -- since Google was coming up empty

The logs, when you enable `--ssl-verbose` will look like this:

```
[uwsgi-ssl] client certificate verify error: num=2:unable to get issuer certificate:depth=1:/C=US/ST=District of Columbia/L=Washington/O=Bike Shed/CN=Root CA 1
139755092575944:error:14089086:SSL routines:ssl3_get_client_certificate:certificate verify failed:s3_srvr.c:3322:
139755092575944:error:140940E5:SSL routines:ssl3_read_bytes:ssl handshake failure:s3_pkt.c:1217:
139755092575944:error:140E0197:SSL routines:SSL_shutdown:shutdown while in init:ssl_lib.c:1077:
[uwsgi-ssl] client certificate verify error: num=2:unable to get issuer certificate:depth=1:/C=US/ST=District of Columbia/L=Washington/O=Bike Shed/CN=Root CA 1
139755092575944:error:14089086:SSL routines:ssl3_get_client_certificate:certificate verify failed:s3_srvr.c:3322:
```

**Comment by funkybob** [comment]
Wasn't this fixed by  #1564  ?

**Comment by paultag** [comment]
Yep!
